src/cilium_manifests.py

```diff
@@ -2,14 +2,31 @@
 
 from typing import Dict
 
-from ops.manifests import ConfigRegistry, ManifestLabel, Manifests
+from ops.manifests import ConfigRegistry, ManifestLabel, Manifests, Patch
+
+
+class SetIPv4CIDR(Patch):
+    """Configure IPv4 CIDR and Node Mask."""
+
+    def __call__(self, obj) -> None:
+        """Update ConfigMap IPv4 CIDR and Mask size."""
+        if not (obj.kind == "ConfigMap" and obj.metadata.name == "cilium-config"):
+            return
+
+        data = obj.data
+        data["cluster-pool-ipv4-cidr"] = self.manifests.config["cluster-pool-ipv4-cidr"]
+        data["cluster-pool-ipv4-mask-size"] = self.manifests.config["cluster-pool-ipv4-mask-size"]
```

Comment:
are there any special relations to these two config options in cilium?

it's interesting that `cluster-pool-ipv4-cidr` includes the subnet mask.  

`10.0.0.0/8` -- `10.0.0.0 network with a 8-bit subnet mask`

Does the `cluster-pool-ipv4-mask-size` not have to match?  Or this is a per-running-kubelet setting?

Do the defaults of `10.0.0.0/8` and `24` mean that each worker node can allocate 256 addresses on the `10.0.0.0/8 ` subnet.  This would mean you could have 65536 (2^16) worker nodes.  Am i understanding this configuration correctly?

Reply:
> Do the defaults of `10.0.0.0/8` and `24` mean that each worker node can allocate 256 addresses on the `10.0.0.0/8 ` subnet. This would mean you could have 65536 (2^16) worker nodes. Am i understanding this configuration correctly?
Correct, the cluster-scope IPAM assigns CIDRs to each node from the cluster-wide pool address space using a host-scope allocator on each node.

You can see how it works [here](https://docs.cilium.io/en/v1.12/concepts/networking/ipam/cluster-pool/).

Reply:
so do these defaults reason well?  

mask-size: `24`  - the subnet mask per node
CIDR: `10.0.0.0/8` -- you can break this into 2^16 (65536) `/24` subnets-per-node

does this indicate the defaults are for us to have 64k nodes in a cluster?  Are these just propagating the upstream defaults?

Reply:
Yes, I just left the defaults from upstream. Do you think we should define our own default values?

Reply:
Okay!  Those defaults come from upstream.  Sweet.  thanks!
Can you update the config.yaml to indicate where the defaults come from or documentation on how to set them?